lesson5.rb

```diff
@@ -0,0 +1,32 @@
+
+#print size of each string in the array
+names.each {|index| print "#{index.size} "}
+print "\n"
+
+#print size of all strings in the array
+size_sum = 0
+names.each do |index|
+	size_sum = size_sum + index.size
+end
+puts "Sum of all strings in the array: #{size_sum}"
+
+#make all the selected strings downcase
+names.map(&:downcase!)
+
+#using the last array, create a hash with key: name and value: name size
+names_hash = {}
+names.map {|index| names_hash[index] = index.size}
+
+#using the hash, get the longest name (this can be done without if using hash and array
+#methods, use the links from the previous slide)
+longest_name = names_hash.keys.sort{|a,b| a.size <=> b.size}.reverse!.first
```

Comment:
You can try the `max_by` method

Reply:
Thank you!
Yes, max_by looks more readable and works faster.